pkg/cli/mirror/catalog_images.go

```diff
@@ -195,22 +209,53 @@ func (o *MirrorOptions) processCatalogRefs(ctx context.Context, catalogsByImage
 
 		klog.Infof("Rendering catalog image %q with file-based catalog ", refExact)
 
-		add, err := builder.LayerFromPath("/configs", filepath.Join(artifactDir, config.IndexDir, "index.json"))
+		configLayerToAdd, err := builder.LayerFromPath("/configs", filepath.Join(artifactDir, config.IndexDir, "index.json"))
 		if err != nil {
 			return fmt.Errorf("error creating add layer: %v", err)
 		}
 
 		// Since we are defining the FBC as index.json,
 		// remove anything that may currently exist
-		deleted, err := deleteLayer("/.wh.configs")
+		deletedConfigLayer, err := deleteLayer("/.wh.configs")
 		if err != nil {
 			return fmt.Errorf("error creating deleted layer: %v", err)
 		}
 
-		// Delete must be first in the slice
-		// so that the /configs directory is deleted
-		// and then add back with the new FBC.
-		layers := []v1.Layer{deleted, add}
+		// white out layer /tmp
+		// TODO : any label in the manifests that might indicate more accurately where the cache is?
+		deletedCacheLayer, err := deleteLayer("/.wh.tmp")
+		if err != nil {
+			return fmt.Errorf("error creating deleted cache layer: %v", err)
+		}
+
+		opmCmdPath, err := findOpmCmd(artifactDir)
+		if err != nil {
+			return fmt.Errorf("cannot find opm in the extracted catalog %v for %s on %s: %v", ctlgRef, runtime.GOOS, runtime.GOARCH, err)
+		}
+		absConfigPath, err := filepath.Abs(filepath.Join(artifactDir, config.IndexDir))
+		if err != nil {
+			return fmt.Errorf("error getting absolute path for catalog's index %v: %v", filepath.Join(artifactDir, config.IndexDir), err)
+		}
+		absCachePath, err := filepath.Abs(filepath.Join(artifactDir, config.TmpDir))
+		if err != nil {
+			return fmt.Errorf("error getting absolute path for catalog's cache %v: %v", filepath.Join(artifactDir, config.TmpDir), err)
+		}
+		//TODO call opm serve /configs –-cache-dir /tmp/cache –-cache-only
+		cmd := exec.Command(opmCmdPath, "serve", absConfigPath, "--cache-dir", absCachePath, "--cache-only")
```

Comment:
In slack you mentioned that Joe said you should do this:

> I think you’ll want to run opm serve <configDir> --cache-dir=<cacheDir> --cache-only --cache-enforce-integrity 

This seems to be missing the `--cache-enforce-integrity`

Reply:
TODO can be removed from the comment I think

Reply:
Done